### PR TITLE
Always send outcome when LTI data present and write tests for assessm…

### PR DESCRIPTION
**Pull request by ajhaupt7**
…ent_results
- Previously scores were not sent if the lti_role was 'admin'
- Changed method to always send outcome if all the lti data is present
- Wrote tests for assessment_result model

Testing:
- run rspec spec/models/user_assessment_spec.rb
- still need to add tests for send_outcome_to_tool_consumer! method

### CODE REVIEW

File: spec/models/assessment_result_spec.rb [hunk 1]
```diff
@@ -0,0 +1,130 @@
+require 'rails_helper'
+
+describe AssessmentResult do
+
+	before do
```
Comment (bracken) [inline spec/models/assessment_result_spec.rb]: Lots of repeated stuff, I'd clean it up just a little bit:

```
  def create_assessment_result(user_assessment, score=100.0, attempt=0)
    create(:assessment_result,
           :external_user_id => user_assessment.eid,
           :score => score,
           :attempt => attempt,
           :lti_outcome_data => {
                   :lti_role => 'admin',
                   :lis_user_id => user_assessment.eid,
                   :lis_result_sourcedid => '',
                   :lis_outcome_service_url => 'https://example.com/api/ltigrade_passback'
           },
           :user_assessment_id => user_assessment.id,
           :session_status => 'pendingLtiOutcome'
    )
  end

  before do
    @user_assessment = create(:user_assessment)
    @assessment_result_1 = create_assessment_result(@user_assessment, 33.0, 0)
    @assessment_result_2 = create_assessment_result(@user_assessment, 67.0, 1)
    @assessment_result_3 = create_assessment_result(@user_assessment, 16.0, 2)
  end
```

File: spec/models/assessment_result_spec.rb [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		@user_assessment = create(:user_assessment)
```
Comment (bracken) [inline spec/models/assessment_result_spec.rb]: It looks like you have you editor set to 4-space indents, can you reconfigure it to 2 spaces?

File: spec/models/assessment_result_spec.rb [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+	context 'post_lti_outcome' do
+		it 'should return true and change session_status to final if should_send_lti_outcome returns false or nil' do
+	  	@assessment_result_2.assessment.update(:kind => 'formative')
+			expect(@assessment_result_2.post_lti_outcome!).to eq true
+			expect(@assessment_result_2.session_status).to eq 'final'
+		end
+	end
+
+	context 'send_outcome_to_tool_consumer!' do
```
Comment (bracken) [inline spec/models/assessment_result_spec.rb]: it's generally preferred to just leave this section out if it's all comments.
If you want to indicate tests that _need_ to be written, you can leave empty it clauses like:

```
it "should send the LTI outcome to the tool consumer"
```

**Comment by JuliannaGreen** [comment]
Looks good!

**Pull request closed without merging**